Introduce the new user-facing Logs API as a placeholder (#4236)
open-telemetry/opentelemetry-specification: specification/logs/event-sdk.md

All implementations of the OpenTelemetry API MUST provide an SDK.

## Instrumentation API Development

> Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).

Reviewer: ```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

> Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).

Reviewer 2: ```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

> The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.

Reviewer: ```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```

> The intent is that this new API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
> 
> No further work is scheduled for the current Event SDK at this time.

Reviewer 3: ```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.

No further work is scheduled for the current Events SDK at this time.
```

Author: Let me adjust this in all references.

Author: I'm going to call it Instrumentation API for consistency though.

All implementations of the OpenTelemetry API MUST provide an SDK.

## Logs Instrumentation API Development

> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#logs-instrumentation-api).

> The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.

Author: This note is for anyone reading it between now and when we have the new Logs API functionality specified.

I am aware that there is active work occurring (as this has existed for several years) and there are some instrumentations that are using this today, I believe in JS, Java (Android) and potentially now in Python (for LLM).

I'm also trying to explicitly defer the exact approach that we take until we get to that point, so by using these enumerations it (hopefully) is also acting as a further warning to not take any dependency on it (if they have not already) and for anyone how has we will hopefully get feedback.
